TheMET/TheMET/ViewControllers/CatalogViewController.swift

```diff
@@ -86,19 +86,29 @@ class CatalogViewController: UIViewController {
                 completion(nil)
                 return
             }
-            self?.loadCatalogCellDataList(departments: responce.departments) { catalogCellDataList in
+            var catalogCellDataList: [CatalogCellData] = []
+       for department in responce.departments {
+            let catalogCellData: CatalogCellData = CatalogCellData(departmentId: department.id, departmentData: .placeholder)
+           catalogCellDataList.append(catalogCellData)
+        }
+            self?.contentStatus = .loaded(catalogCellDataList)
+            self?.loadCatalogCellDataList(departments: responce.departments, catalogCellDataList: catalogCellDataList) { catalogCellDataList in
                 completion(catalogCellDataList)
             }
         }
     }
     
-    private func loadCatalogCellDataList(departments: [Department], completion: @escaping ([CatalogCellData]) -> Void) {
+    private func loadCatalogCellDataList(departments: [Department], catalogCellDataList: [CatalogCellData], completion: @escaping ([CatalogCellData]) -> Void) {
         let group = DispatchGroup()
-        var catalogCellDataList:[CatalogCellData] = []
         for department in departments {
             group.enter()
-            self.loadCatalogCellData(department: department, completion: { cellData in
-                catalogCellDataList.append(cellData)
+            self.loadCatalogCellData(department: department, completion: { [weak self] cellData in
```

Comment:
There is no need in using `DispatchGroup` anymore as all the loadings should not be synchronised anymore

Reply:
+